pkg/bbgo/margin_info_updater.go

```diff
@@ -0,0 +1,91 @@
+package bbgo
+
+import (
+	"context"
+	"time"
+
+	"github.com/c9s/bbgo/pkg/fixedpoint"
+	"github.com/c9s/bbgo/pkg/types"
+	"github.com/c9s/bbgo/pkg/util/timejitter"
+)
+
+type MaxBorrowableCallbackType func(asset string, amount fixedpoint.Value)
+
+//go:generate callbackgen -type MarginInfoUpdater
+type marginInfoUpdaterType struct {
+	BorrowRepayService types.MarginBorrowRepayService
+
+	assets                 map[string]struct{}
+	maxBorrowableCallbacks []MaxBorrowableCallbackType
+}
+
+func NewMarginInfoUpdaterFromSession(
+	session *ExchangeSession,
+) *marginInfoUpdaterType {
+	if !session.Margin || session.Exchange == nil {
+		return nil
+	}
```

Comment:
ditto

Reply:
Just like what I reply above, I'd rather keep the check here.

Reply:
fixed.